source/grid/grid_generator.cc

```diff
@@ -9685,6 +9685,139 @@ namespace GridGenerator
         AssertThrow(false, ExcNotImplemented());
       }
   }
+
+
+
+  template <>
+  void
+  uniform_channel_with_sphere(
+    Triangulation<3>                &tria,
+    const std::vector<unsigned int> &lengths_and_heights,
+    const unsigned int               depth,
+    const bool                       colorize)
+  {
+    const double radius = 0.5;
+    const double box_radius = 1;
```

Comment:
The depth default value is equal to one. Are there limitations on the depth value with respect to the other parameter? Is the depth the number of subdivision in the depth or the physical depth in terms of number of sphere radius? :)